source/patterns/@aws-solutions-constructs/core/lib/elasticsearch-helper.ts

```diff
@@ -257,7 +257,11 @@ export function GetSubnetsAndSecurityGroups(lambdaFunction: lambda.Function, vpc
     }
 
     const securityGroupIds: string[] = [];
-    lambdaFunction.connections.securityGroups.forEach(element => securityGroupIds.push(element.securityGroupId));
+
+    // Get the associated security group ids otherwise use the default security group ids
+    if (lambdaFunction) {
```

Comment:
This isn't addressing my issue. The problem is that this implementation assumes the client of the domain is a Lambda function - eg it assumes this is part of aws-lambda-elasticsearch.

What's the objective here? Does the Domain need to allow access from these Security Groups? Does our code use these security groups to set up security groups for the domain?

It seems to me the better approach would be to create a function in lambda-helper.ts that extracts the security groups for a Lambda function, call that from index.ts and send the resulting security group collection in the props for buildDomain. This is not a fully formed idea, I'm throwing it out there for you to turn into fully formed coded.